packages/notifi-dapp-example/src/context/InjectiveWalletContext.tsx

```diff
@@ -0,0 +1,249 @@
+    useState<PhantomWalletKeys | null>(null);
+  const selectWallet = (wallet: keyof InjectiveWallets | null) => {
+    setSelectedWallet(wallet);
+  };
+
+  const [error, setError] = useState<Error | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const connectLeapWallet = async () => {
+    if (!injectiveLeapWallet.strategies.leap) {
+      return null;
+    }
+    setIsLoading(true);
+    try {
+      injectiveLeapWallet.setWallet(Wallet.Leap);
+      const key = await injectiveLeapWallet.getPubKey();
+      const accounts = await injectiveLeapWallet.getAddresses();
+      const walletKeys = {
+        bech32: accounts[0],
+        base64: Buffer.from(key).toString('base64'),
+      };
+      selectWallet('leap');
+      setWalletKeysLeap(walletKeys);
+      setWalletKeysToLocalStorage('leap', walletKeys);
+      setIsLoading(false);
+      return walletKeys;
+    } catch (e) {
+      throwError(
+        new Error('Leap connection failed, check console for details'),
+      );
+      console.error(e);
+    }
+    setIsLoading(false);
+    return null;
+  };
+
+  const connectPhantomWallet = async () => {
+    if (!window.solana) {
+      return null;
+    }
+    if (!injectivePhantomWallet.strategies.phantom) {
+      return null;
+    }
+    setIsLoading(true);
+    try {
+      injectivePhantomWallet.setWallet(Wallet.Phantom);
+      const accounts = await injectivePhantomWallet.getAddresses();
+      const walletKeys = {
+        bech32: converter('inj').toBech32(accounts[0]),
+        base64: Buffer.from(accounts[0]).toString('base64'),
+      };
+      selectWallet('phantom');
+      setWalletKeysPhantom(walletKeys);
+      setWalletKeysToLocalStorage('phantom', walletKeys);
+      setIsLoading(false);
+      return walletKeys;
+    } catch (e) {
+      throwError(
+        new Error('Leap connection failed, check console for details'),
+      );
+      console.error(e);
+    }
+    setIsLoading(false);
+    return null;
+  };
+
+  const leapSignArbitrary = useCallback(
+    async (message: string) => {
+      if (!walletKeysLeap || !injectiveLeapWallet.strategies.leap) {
+        return;
+      }
+      setIsLoading(true);
+      try {
+        const result = await injectiveLeapWallet.signArbitrary(
+          walletKeysLeap.bech32,
+          message,
+        );
+        return result;
+      } catch (e) {
+        throwError(
+          new Error('Leap signArbitrary failed, check console for details'),
+        );
+        console.error(e);
+      }
+      setIsLoading(false);
+    },
+    [walletKeysLeap],
+  );
+
+  const phantomSignArbitrary = useCallback(
+    async (message: string) => {
+      if (!walletKeysPhantom || !injectivePhantomWallet.strategies.phantom) {
+        return;
+      }
+      setIsLoading(true);
+      try {
+        const result = await injectivePhantomWallet.signArbitrary(
+          walletKeysPhantom.bech32,
+          message,
```

Comment:
Just FYI, in metamask, we convert the message to hex by `Buffer.from(message).toString('hex')`. 
But I tried doing so, the `signiture not match` error is still happening.

Reply:
Should we convert message to hex for phantom?

Reply:
No, I think the `getAddresses` from the `walletStrategy` is already an array of hex string